### PR TITLE
[BugFix] fix tableFunction return wrong RowOutputInfo

### LINKED ISSUE #19232: Query fails with "cannot find statistics of col"
### Steps to reproduce the behavior (Required)
```
CREATE DATABASE TEST;

CREATE TABLE `TEST`.`ACTIVITY` (`case` BIGINT, `timestamp` DATETIME, `ID_activity` BIGINT, `activity` BIGINT, `activityname` STRING, `FK_8_activity_case` BIGINT) PRIMARY KEY(`case`,`timestamp`,`ID_activity`) DISTRIBUTED BY HASH (`case`) PROPERTIES("replication_num" = "1");
CREATE TABLE `TEST`.`ORDERPOS` (`ID_orderpos` BIGINT, `orderpos` BIGINT, `case` BIGINT, `orderid` STRING, `pos` BIGINT, `country` STRING, `city` STRING, `FK_8_orderpos_case` BIGINT)  DISTRIBUTED BY HASH (`ID_orderpos`) PROPERTIES("replication_num" = "1");
CREATE TABLE `TEST`.`CASE` (`ID_case` BIGINT, `case` BIGINT, `orderid` STRING, `FK_8_activity_case` BIGINT, `FK_8_orderpos_case` BIGINT, `PU_9_ARRAY_AGG_8_activity_case` ARRAY<BIGINT>, `PU_9_ARRAY_AGG_8_activity_timestamp` ARRAY<DATETIME>, `PU_9_ARRAY_AGG_8_activity_ID_activity` ARRAY<BIGINT>, `PU_9_ARRAY_AGG_8_activity_activity` ARRAY<BIGINT>, `PU_9_ARRAY_AGG_8_activity_activityname` ARRAY<STRING>, `PU_9_ARRAY_AGG_8_activity_FK_8_activity_case` ARRAY<BIGINT>)  DISTRIBUTED BY HASH (`ID_case`) PROPERTIES("replication_num" = "1");
```

```
WITH `CASE` AS (
  SELECT
    `FK_8_ACTIVITY_CASE`,
    `FK_8_ORDERPOS_CASE`,
    `PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME`,
    `PU_9_ARRAY_AGG_8_ACTIVITY_CASE`,
    `PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE`,
    `ORDERID`
  FROM
    `TEST`.`CASE`
),
`ORDERPOS` AS (
  SELECT
    `FK_8_ORDERPOS_CASE`,
    (
      (`COUNTRY` <=> ("DE"))
      OR (`COUNTRY` <=> ("FR"))
    ) AS `ParserContext_FilterBooleanOrNull_017`,
    `COUNTRY`,
    `ORDERID`
  FROM
    `TEST`.`ORDERPOS`
),
`ParserContext_Table_CASE_000` AS (
  SELECT
    `CASE`.`FK_8_ORDERPOS_CASE` AS `JoinColumnResolver_Column_FK_8_ORDERPOS_CASE_004`,
    `CASE`.`PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027`,
    `CASE`.`PU_9_ARRAY_AGG_8_ACTIVITY_CASE` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025`,
    `CASE`.`PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011`,
    `CASE`.`ORDERID` AS `JoinColumnResolver_Column_ORDERID_005`
  FROM
    `CASE`
),
`TableScanSequence_ORDERPOS_008` AS (
  SELECT
    `ORDERPOS`.`FK_8_ORDERPOS_CASE` AS `JoinColumnResolver_Column_FK_8_ORDERPOS_CASE_013`,
    `ParserContext_Table_CASE_000`.`JoinColumnResolver_Column_ORDERID_005` AS `JoinColumnResolver_Column_ORDERID_006`,
    `ORDERPOS`.`ORDERID` AS `JoinColumnResolver_Column_ORDERID_007`
  FROM
    `ORDERPOS`
    LEFT JOIN `ParserContext_Table_CASE_000` ON (
      `ORDERPOS`.`FK_8_ORDERPOS_CASE` = `ParserContext_Table_CASE_000`.`JoinColumnResolver_Column_FK_8_ORDERPOS_CASE_004`
    )
),
`ParserContext_Table_ORDERPOS_003` AS (
  SELECT
    `JoinColumnResolver_Column_FK_8_ORDERPOS_CASE_013`,
    `JoinColumnResolver_Column_ORDERID_006`,
    `JoinColumnResolver_Column_ORDERID_007`,
    (
      (
        `JoinColumnResolver_Column_ORDERID_006` <=> `JoinColumnResolver_Column_ORDERID_007`
      )
    ) AS `ParserContext_FilterBooleanOrNull_002`
  FROM
    `TableScanSequence_ORDERPOS_008`
),
`ParserContext_Flattened_001` AS (
  SELECT
    TableScan_UNNEST.JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027 AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027`,
    TableScan_UNNEST.JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025 AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025`,
    TableScan_UNNEST.JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011 AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011`
  FROM
    `ParserContext_Table_CASE_000`,
    UNNEST(
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011,
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025,
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027
    ) AS TableScan_UNNEST(
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011,
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025,
      JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027
    )
),
`TableScanSequence_Flattened_010` AS (
  SELECT
    `ParserContext_Flattened_001`.`JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_027` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_028`,
    `ParserContext_Flattened_001`.`JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_025` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_026`,
    `ParserContext_Flattened_001`.`JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_011` AS `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012`
  FROM
    `ParserContext_Flattened_001`
),
`TableScanSequence_Flattened_016` AS (
  SELECT
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_028`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_026`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012`,
    EXISTS (
      SELECT
        *
      FROM
        `CASE`,
        `ParserContext_Table_ORDERPOS_003`
      WHERE
        `TableScanSequence_Flattened_010`.`JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012` = `CASE`.`FK_8_ACTIVITY_CASE`
        AND `ParserContext_Table_ORDERPOS_003`.`JoinColumnResolver_Column_FK_8_ORDERPOS_CASE_013` = `CASE`.`FK_8_ORDERPOS_CASE`
        AND `ParserContext_Table_ORDERPOS_003`.`ParserContext_FilterBooleanOrNull_002`
    ) AS `ParserContext_FilterBoolean_014`
  FROM
    `TableScanSequence_Flattened_010`
),
`TableScanSequence_Flattened_019` AS (
  SELECT
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_028`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_026`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012`,
    `ParserContext_FilterBoolean_014`
  FROM
    `TableScanSequence_Flattened_016`
),
`TableScanSequence_Flattened_020` AS (
  SELECT
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_028`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_026`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012`,
    `ParserContext_FilterBoolean_014`
  FROM
    `TableScanSequence_Flattened_019`
),
`ParserContext_Table_Flattened_009` AS (
  SELECT
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_ACTIVITYNAME_028`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_CASE_026`,
    `JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012`,
    `ParserContext_FilterBoolean_014`,
    EXISTS (
      SELECT
        *
      FROM
        `CASE`,
        `ORDERPOS`
      WHERE
        `TableScanSequence_Flattened_020`.`JoinColumnResolver_Column_PU_9_ARRAY_AGG_8_ACTIVITY_FK_8_ACTIVITY_CASE_012` = `CASE`.`FK_8_ACTIVITY_CASE`
        AND `ORDERPOS`.`FK_8_ORDERPOS_CASE` = `CASE`.`FK_8_ORDERPOS_CASE`
        AND `ORDERPOS`.`ParserContext_FilterBooleanOrNull_017`
    ) AS `ParserContext_FilterBoolean_021`
  FROM
    `TableScanSequence_Flattened_020`
)
SELECT * FROM ParserContext_Table_Flattened_009;
```

### Expected behavior (Required)
Query should succeed without errors

### Real behavior (Required)
```
cannot find statistics of col: 321: joincolumnresolver_column_pu_9_array_agg_8_activity_fk_8_activity_case_011
```

### StarRocks version (Required)
3.0

### PR DESCRIPTION
**Pull request by packy92**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #19232

## Problem Summary(Required) ：

Forget to implment deriveRowOutputInfo correctly in LogicalTableFunctionOperator.
When use `JoinLeftAsscomRule` transform plan, 
```
RowOutputInfo newBotJoinRowInfo = deriveBotJoinRowInfo(newBotJoinColSet, newBotJoinLeftChild,
                newBotJoinRightChild, splitter);
```
We use the bottom child row output info to build the  newBotJoinRowInfo. If one child return wrong rowOutputInfo, the transformed plan is invalid. 

Add `List<ColumnRefOperator> fnResultColumns`, `List<ColumnRefOperator> outerColumns` field in both Logical and Physical TableFunctionOperator to derive correct rowOutputInfo and it will also helps to validate col type in the future.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 54 / 58 (93.10%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalTreeAnchorOperator.java|0|1|00.00%|[43]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/MockOperator.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalApplyOperator.java|9|11|81.82%|[152, 153]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToSQLBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeApplyWithTableFunction.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RowOutputInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/TableFunctionImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalTableFunctionOperator.java|24|24|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneTableFunctionColumnRule.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalTableFunctionOperator.java|5|5|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalTableFunctionOperator.java [hunk 1]
```diff
@@ -32,40 +34,57 @@
 
 public class LogicalTableFunctionOperator extends LogicalOperator {
     private final TableFunction fn;
-    //ColumnRefSet represent output by table function
+
+    // Table function output cols. It's used to validate plan.
+    private final List<ColumnRefOperator> fnResultColumns;
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalTableFunctionOperator.java:39]: The cost may be small.

[the remark this answers is not in the document]
Comment (packy92) [reply fe/fe-core/src/main/java/com/starrocks/sql/optimizer/operator/logical/LogicalTableFunctionOperator.java]: fixed. please review it again.

**Review by kangkaisen — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by packy92** [comment]
run starrocks_fe_unittest

**Comment by packy92** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 83 / 94 (88.30%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalTreeAnchorOperator.java|0|1|00.00%|[43]|
|:large_blue_circle:|com/starrocks/sql/Explain.java|0|2|00.00%|[578, 579]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/MockOperator.java|0|1|00.00%|[63]|
|:large_blue_circle:|com/starrocks/sql/optimizer/statistics/StatisticsCalculator.java|3|4|75.00%|[1073]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/physical/PhysicalTableFunctionOperator.java|11|14|78.57%|[103, 104, 105]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalApplyOperator.java|9|11|81.82%|[152, 153]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/mv/MaterializedViewRewriter.java|9|10|90.00%|[210]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|5|5|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToSQLBuilder.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/MergeApplyWithTableFunction.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/RowOutputInfo.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/implementation/TableFunctionImplementationRule.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/operator/logical/LogicalTableFunctionOperator.java|28|28|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PruneTableFunctionColumnRule.java|7|7|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/optimizer/rule/transformation/PushDownPredicateTableFunctionRule.java|2|2|100.00%|[]|

**Comment by packy92** [comment]
run starrocks_admit_test

**Review by kangkaisen — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**